tangram_bundler/__init__.py

```diff
@@ -1,97 +1,231 @@
-    if 'textures' in yaml_file:
-        for texture in yaml_file['textures']:
-            if 'url' in yaml_file['textures'][texture]:
-                file_list.append(os.path.normpath(folder + '/' + yaml_file['textures'][texture]['url']))
-
-    # Search for u_envmap or u_ramp urls
-    if 'styles' in yaml_file:
-        for style in yaml_file['styles']:
-            if 'shaders' in yaml_file['styles'][style]:
-                if 'uniforms' in yaml_file['styles'][style]['shaders']:
-                    for uniform in yaml_file['styles'][style]['shaders']['uniforms']:
-                        addUniformTextureDependency(file_list, folder, yaml_file, style, uniform)
-
-    # Search for inner dependencies
-    if 'import' in yaml_file:
-        if (type(yaml_file['import']) is str):
-            addDependencies(file_list, folder + '/' + yaml_file['import'])
+    if 'textures' in rootNode:
+        texturesNode = rootNode['textures']
+        for texture in texturesNode:
+            textureNode = texturesNode[texture]
+            if 'url' in textureNode:
+                fileList.append(os.path.relpath(textureNode['url'], basePath))
+
+    # Search for asset url in styles
+    if 'styles' in rootNode:
+        stylesNode = rootNode['styles']
+        for style in stylesNode:
+            styleNode = stylesNode[style]
+            if 'shaders' in styleNode:
+                shadersNode = styleNode['shaders']
+                if 'uniforms' in shadersNode:
+                    uniformsNode = shadersNode['uniforms']
+                    for uniform in uniformsNode:
+                        addUniformTextureDependency(fileList, basePath, rootNode, style, uniform)
+            if 'texture' in styleNode:
+                fileList.append(os.path.relpath(styleNode['texture'], basePath))
+            if 'material' in styleNode:
+                materialNode = styleNode['material']
+                if (type(materialNode) is dict):
+                    for prop in ['emission', 'ambient', 'diffuse', 'specular', 'normal']:
+                        propNode = materialNode[prop]
+                        if (type(propNode) is dict and 'texture' in propNode):
+                            fileList.append(os.path.relpath(propNode['texture'], basePath))
+
+def ignoreFileForBundling(fileName):
+    # we atleast know we need to ignore bundling any network url (todo: fetch yaml http urls and do not ignore these)
+    if (fileName.startswith("http") and not fileName.endswith(".yaml")):
+        return False
+    return True
+
+def collectAllImports(allImports):
+    if (len(allImports) <= 0):
+        return
+    
+    subImports = {}
+    for fileName in allImports:
+        yamlFile = allImports[fileName]
+        directory = os.path.dirname(fileName)
+        if 'import' in yamlFile:
+            importNode = yamlFile['import']
+            if (type(importNode) is str):
+                importPath = os.path.abspath(directory + '/' + importNode)
```

Comment:
Is `os.path.join` a better way to do this? We should probably be using `os.sep` instead of `'/'` at least.

Reply:
Oh or does `urljoin` work for this?